components/sup/src/manager/mod.rs

```diff
@@ -413,15 +461,107 @@ impl Manager {
                     }
                     Err(err) => warn!("Error loading service config after update, err={}", err),
                 }
-                self.state.butterfly.insert_service(rumor);
+                self.butterfly.insert_service(rumor);
                 last_update.service_counter += 1;
             }
         }
     }
 
+    fn persist_state(&self) {
+        debug!("Writing census state to disk");
+        self.persist_census_state();
+        debug!("Writing butterfly state to disk");
+        self.persist_butterfly_state();
+        debug!("Writing services state to disk");
+        self.persist_services_state();
+    }
+
+    fn persist_census_state(&self) {
+        let tmp_file = self.fs_cfg.census_data_path.with_extension("dat.tmp");
+        let file = match OpenOptions::new()
```

Comment:
Ah yeah, I totally can just use that instead here - I was pulling this from another piece of code that I wrote when I also wanted to be able to read the file. Just swapped it to `File::create`!